internal/controller/builder/builder.go

```diff
@@ -775,26 +775,24 @@ func BuildVolumeSnapshotClass(name string, driver string) (*snapshotv1.VolumeSna
 }
 
 func BuildServiceAccount(cluster *appsv1alpha1.Cluster) (*corev1.ServiceAccount, error) {
-	const tplFile = "rbac_template.cue"
-
-	sa := &corev1.ServiceAccount{}
-	if err := buildFromCUE(tplFile, map[string]any{
-		"cluster": cluster,
-	}, "serviceaccount", sa); err != nil {
-		return nil, err
-	}
-	return sa, nil
+	return buildRBACObject[corev1.ServiceAccount](cluster, "serviceaccount")
 }
 
 func BuildRoleBinding(cluster *appsv1alpha1.Cluster) (*rbacv1.RoleBinding, error) {
-	const tplFile = "rbac_template.cue"
+	return buildRBACObject[rbacv1.RoleBinding](cluster, "rolebinding")
+}
 
-	rb := &rbacv1.RoleBinding{}
-	if err := buildFromCUE(tplFile, map[string]any{
-		"cluster": cluster,
-	}, "rolebinding", rb); err != nil {
+func BuildClusterRoleBinding(cluster *appsv1alpha1.Cluster) (*rbacv1.ClusterRoleBinding, error) {
+	return buildRBACObject[rbacv1.ClusterRoleBinding](cluster, "clusterrolebinding")
+}
+
+func buildRBACObject[Tp corev1.ServiceAccount | rbacv1.RoleBinding | rbacv1.ClusterRoleBinding](
+	cluster *appsv1alpha1.Cluster, key string) (*Tp, error) {
+	const tplFile = "rbac_template.cue"
+	var obj Tp
+	pObj := &obj
+	if err := buildFromCUE(tplFile, map[string]any{"cluster": cluster}, key, pObj); err != nil {
```

Comment:
Maybe you could use reflect.TypeOf(Tp) to remove parameter 'key string'.

Reply:
The key is now just as a unique string, using reflect.TypeOf means that the string in CUE file has type semantics, that would be too tightly coupled.